dso/model.py

```diff
@@ -22,56 +24,94 @@ class ScanArtifact:
 
 class Datasource:
     WHITESOURCE = 'whitesource'
-    PROTECODE = 'protecode'
+    BDBA = 'bdba' # formerly protecode
     CHECKMARX = 'checkmarx'
     CLAMAV = 'clamav'
+    CC_UTILS = 'cc-utils'
 
 
-@dataclasses.dataclass
+@dataclasses.dataclass(frozen=True)
+class Artefact:
+    artefact_name: str
+    artefact_version: str
+    artefact_type: str
+    artefact_extra_id: dict
+
+
+@dataclasses.dataclass(frozen=True)
 class ArtefactReference:
```

Comment:
having both `Artefact` and `ArtefactReference` is rather misleading, considering `Artefact` is, in fact, an artefact reference, whereas `ArtefactReference` is a "qualified" artefact reference.

Rather call them, maybe: LocalArtefactId, and ComponentArtefactId (or -Reference, if you so prefer)